[PDI-14721] Problem with MySQL and BLOBs - ValueMetaInterface.TYPE_BINARY is being set to ValueMetaInterface.STRING

AliaksandrShuhayeu opened the pull request:
-there was deleted setting wrong type

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java
  core/src/test/java/org/pentaho/di/core/database/DatabaseUnitTest.java
  core/src/test/java/org/pentaho/di/core/row/value/ValueMetaBaseTest.java
  engine/src/main/java/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java
```

Newly Broken Tests:
======================
###### org.pentaho.di.trans.steps.databaselookup.DatabaseLookupUTest.mySqlVariantDbIsLazyConverted:

:red_circle: `org.mockito.exceptions.verification.junit.ArgumentsAreDifferent`

```java
org.mockito.exceptions.verification.junit.ArgumentsAreDifferent: 

Argument(s) are different! Wanted:
database.getLookup(
    <any>,
    <any>,
    true
);
-> at org.pentaho.di.trans.steps.databaselookup.DatabaseLookupUTest.mySqlVariantDbIsLazyConverted(DatabaseLookupUTest.java:108)
Actual invocation has different arguments:
database.getLookup(
    Mock for PreparedStatement, hashCode: 175186756,
    false,
    false
);
-> at org.pentaho.di.trans.steps.databaselookup.DatabaseLookupUTest.mySqlVar
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.di.core.row.value.ValueMetaBase
 * Branch Change:  + .16%
 * Complexity Change:  + .10%
 * Instruction Change: **-.12%**:small_red_triangle_down:
 * Line Change: **-.29%**:small_red_triangle_down:

###### org.pentaho.di.core.injection.bean.BeanInjector
 * Branch Change:  + .84%
 * Complexity Change:  + 1.39%

###### org.pentaho.di.core.lifecycle.KettleLifecycleSupport
 * Branch Change: **-8.33%**:small_red_triangle_down:
 * Complexity Change: **-9.09%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.evaluatetablecontent.JobEntryEvalTableContent
 * Branch Change: **-1.01%**:small_red_triangle_down:
 * Complexity Change: **-1.06%**:small_red_triangle_down:
 * Instruction Change: **-.17%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.sql.JobEntrySQL
 * Instruction Change:  + 3.28%
 * Line Change:  + 3.87%

###### org.pentaho.di.job.entries.waitforsql.JobEntryWaitForSQL
 * Branch Change:  + .66%
 * Complexity Change:  + .93%
 * Instruction Change:  + .13%

###### org.pentaho.di.trans.step.StepMeta
 * Branch Change:  + .68%
 * Complexity Change:  + .58%
 * Instruction Change:  + .19%
 * Line Change:  + .26%

###### org.pentaho.di.trans.steps.csvinput.CsvInputMeta
 * Instruction Change:  + .05%

###### org.pentaho.di.trans.steps.databaselookup.DatabaseLookup
 * Branch Change:  + 1.24%
 * Complexity Change:  + .77%
 * Instruction Change:  + .20%
 * Line Change:  + .67%

###### org.pentaho.di.trans.steps.datagrid.DataGridMeta
 * Branch Change:  + 1.92%
 * Complexity Change:  + 1.69%
 * Instruction Change:  + .17%

###### org.pentaho.di.trans.steps.dbproc.DBProcMeta
 * Instruction Change:  + .11%

###### org.pentaho.di.trans.steps.groupby.GroupByMeta
 * Branch Change:  + 2.08%
 * Complexity Change:  + 2.11%
 * Instruction Change:  + .28%
 * Line Change:  + .74%

###### org.pentaho.di.trans.steps.insertupdate.InsertUpdateMeta
 * Instruction Change: **-.05%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.ldapoutput.LDAPOutputMeta
 * Instruction Change:  + .06%

###### org.pentaho.di.trans.steps.memgroupby.MemoryGroupByMeta
 * Branch Change:  + 3.53%
 * Complexity Change:  + 4.05%
 * Instruction Change:  + .18%
 * Line Change:  + .48%

###### org.pentaho.di.trans.steps.processfiles.ProcessFilesMeta
 * Branch Change: **-3.57%**:small_red_triangle_down:
 * Complexity Change: **-2.13%**:small_red_triangle_down:
 * Instruction Change: **-.33%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.replacestring.ReplaceStringMeta
 * Instruction Change:  + .07%

###### org.pentaho.di.trans.steps.singlethreader.SingleThreaderMeta
 * Instruction Change:  + .10%

###### org.pentaho.di.trans.steps.sort.SortRowsMeta
 * Instruction Change:  + .08%

###### org.pentaho.di.trans.steps.synchronizeaftermerge.SynchronizeAfterMergeMeta
 * Instruction Change:  + .04%

###### org.pentaho.di.trans.steps.tableoutput.TableOutputMeta
 * Branch Change:  + .91%
 * Complexity Change:  + .82%

###### org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta
 * Instruction Change: **-.18%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.gpload.GPLoadMeta
 * Instruction Change: **-.13%**:small_red_triangle_down:

###### org.pentaho.di.ui.repository.pur.repositoryexplorer.model.UISecurityRole
 * Branch Change: **-1.79%**:small_red_triangle_down:
 * Complexity Change: **-1.54%**:small_red_triangle_down:
 * Instruction Change: **-.16%**:small_red_triangle_down:
 * Line Change: **-.58%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.salesforceinsert.SalesforceInsertMeta
 * Instruction Change: **-.15%**:small_red_triangle_down:

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  core/src/main/java/org/pentaho/di/core/database/Database.java
  core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java
  core/src/test/java/org/pentaho/di/core/database/DatabaseUnitTest.java
  core/src/test/java/org/pentaho/di/core/row/value/ValueMetaBaseTest.java
  engine/src/main/java/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java
  engine/src/test/java/org/pentaho/di/trans/steps/databaselookup/DatabaseLookupUTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.di.core.database.Database
 * Complexity Change:  + .03%
 * Instruction Change:  + .00%
 * Line Change:  + .02%
 * Method Change:  + .23%

###### org.pentaho.di.core.row.value.ValueMetaBase
 * Instruction Change: **-.16%**:small_red_triangle_down:
 * Line Change: **-.29%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.evalfilesmetrics.JobEntryEvalFilesMetrics
 * Branch Change: **-.39%**:small_red_triangle_down:
 * Complexity Change: **-.51%**:small_red_triangle_down:
 * Instruction Change: **-.08%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.evaluatetablecontent.JobEntryEvalTableContent
 * Branch Change:  + 2.02%
 * Complexity Change:  + 2.13%
 * Instruction Change:  + .34%

###### org.pentaho.di.job.entries.getpop.MailConnectionMeta
 * Branch Change:  + 1.00%
 * Complexity Change:  + 1.41%
 * Instruction Change:  + .30%

###### org.pentaho.di.trans.steps.checksum.CheckSumMeta
 * Branch Change: **-1.35%**:small_red_triangle_down:
 * Complexity Change: **-1.45%**:small_red_triangle_down:
 * Instruction Change: **-.21%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.constant.ConstantMeta
 * Instruction Change: **-.10%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.csvinput.CsvInputMeta
 * Instruction Change: **-.05%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.databaselookup.DatabaseLookup
 * Branch Change:  + 1.24%
 * Complexity Change:  + .77%
 * Instruction Change:  + .20%
 * Line Change:  + .67%

###### org.pentaho.di.trans.steps.dbproc.DBProcMeta
 * Instruction Change:  + .11%

###### org.pentaho.di.trans.steps.dimensionlookup.DimensionLookupMeta
 * Branch Change: **-.31%**:small_red_triangle_down:
 * Instruction Change: **-.06%**:small_red_triangle_down:
 * Line Change:  + .23%
 * Method Change:  + 1.02%

###### org.pentaho.di.trans.steps.groupby.GroupByMeta
 * Branch Change: **-1.04%**:small_red_triangle_down:
 * Complexity Change: **-1.05%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.ifnull.IfNullMeta
 * Branch Change:  + 2.00%
 * Complexity Change:  + 1.89%
 * Instruction Change:  + .17%

###### org.pentaho.di.trans.steps.insertupdate.InsertUpdateMeta
 * Instruction Change: **-.05%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.memgroupby.MemoryGroupByMeta
 * Branch Change:  + 3.53%
 * Complexity Change:  + 4.05%

###### org.pentaho.di.trans.steps.rowgenerator.RowGeneratorMeta
 * Instruction Change: **-.08%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.setvalueconstant.SetValueConstantMeta
 * Instruction Change: **-.14%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.synchronizeaftermerge.SynchronizeAfterMergeMeta
 * Instruction Change: **-.04%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.textfileinput.TextFileInputMeta
 * Branch Change: **-.53%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.googleanalytics.GaInputStepMeta
 * Instruction Change:  + .07%

###### org.pentaho.di.trans.steps.gpload.GPLoadMeta
 * Instruction Change: **-.22%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.salesforceupdate.SalesforceUpdateMeta
 * Instruction Change: **-.17%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.salesforceupsert.SalesforceUpsertMeta
 * Instruction Change:  + .14%

AliaksandrShuhayeu commented:
It is fix for PDI-14721 , ESR-5021 . 
Also I tested PDI-4812 . Current changes don't broke it.
With this fix we don't need part of fix PDI-8373 . My commit contains removing a part of fix PDI-8373 with a workaround. I tested it too. Current changes don't broke PDI-8373 . 
I tested connected PDI-8290 . PPDI-8290 is not duplicate and has a little other case in the transformation. PDI-8373 and PDI-8290 work correct with my changes. 
PDI-8373 has attachment 42321.zip. It doesn't work correct in master but my changes fix it too.
Short.
Before fix (master now): 
PDI-14721 , ESR-5021 There is this issue 
PDI-4812 work correct 
PDI-8290 work correct 
PDI-8373 attachment 42321.zip have a customer problem. It was not fixed.
Current changes: 
PDI-14721 , ESR-5021 , PDI-4812 , PDI-8290 , PDI-8373 work correct

AliaksandrShuhayeu commented:
@mbatchelor Could you please review it?

mbatchelor commented on core/src/main/java/org/pentaho/di/core/database/Database.java:
```diff
@@ -2858,11 +2858,7 @@ public void setProcLookup( String proc, String[] arg, String[] argdir, int[] arg
-    return getLookup( failOnMultipleResults, false );
-  }
-
-  public Object[] getLookup( boolean failOnMultipleResults, boolean lazyConversion ) throws KettleDatabaseException {
```
Why was this method removed ? It's a public method in Database, so it could be called by lots of other things. Don't believe you can do this.

mbatchelor commented on core/src/main/java/org/pentaho/di/core/database/Database.java:
```diff
@@ -2873,17 +2869,12 @@ public void setProcLookup( String proc, String[] arg, String[] argdir, int[] arg
       res = ps.executeQuery();
 
-      Object[] ret = getRow( res, lazyConversion );
+      Object[] ret = getRow( res );
```
Why was the specification of lazy conversion removed here ? I think it needs to be put back.

AliaksandrShuhayeu commented:
All these strings were added in scope of  PDI-8373. All these strings are a workaround in scope of PDI-8373. 
Object[] ret = getRow( res, lazyConversion );
We could get second argument yes only in org.pentaho.di.trans.steps.databaselookup.DatabaseLookup L=141 . I removed it as not actual workaround.
I deleted my changes in Database.java. But these strings are not used in kettle more.

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java
  core/src/test/java/org/pentaho/di/core/database/DatabaseUnitTest.java
  core/src/test/java/org/pentaho/di/core/row/value/ValueMetaBaseTest.java
  engine/src/main/java/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java
  engine/src/test/java/org/pentaho/di/trans/steps/databaselookup/DatabaseLookupUTest.java
```

Newly Broken Tests:
======================
###### org.pentaho.di.trans.steps.databaselookup.DatabaseLookupUTest.mySqlVariantDbIsLazyConverted:

:red_circle: `org.mockito.exceptions.verification.junit.ArgumentsAreDifferent`

```java
org.mockito.exceptions.verification.junit.ArgumentsAreDifferent: 

Argument(s) are different! Wanted:
database.getLookup(<any>, <any>);
-> at org.pentaho.di.trans.steps.databaselookup.DatabaseLookupUTest.mySqlVariantDbIsLazyConverted(DatabaseLookupUTest.java:108)
Actual invocation has different arguments:
database.getLookup(false);
-> at org.pentaho.di.trans.steps.databaselookup.DatabaseLookupUTest.mySqlVariantDbIsLazyConverted(DatabaseLookupUTest.java:106)

	at org.pentaho.di.trans.steps.databa
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.di.core.row.value.ValueMetaBase
 * Instruction Change: **-.16%**:small_red_triangle_down:
 * Line Change: **-.29%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.ftpsget.JobEntryFTPSGet
 * Branch Change:  + .53%
 * Complexity Change:  + .56%
 * Instruction Change:  + .07%

###### org.pentaho.di.job.entries.waitforsql.JobEntryWaitForSQL
 * Branch Change:  + .66%
 * Complexity Change:  + .93%
 * Instruction Change:  + .13%

###### org.pentaho.di.trans.step.StepMeta
 * Branch Change:  + .68%
 * Complexity Change:  + .58%
 * Instruction Change:  + .19%
 * Line Change:  + .26%

###### org.pentaho.di.trans.steps.checksum.CheckSumMeta
 * Instruction Change: **-.11%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.csvinput.CsvInputMeta
 * Instruction Change: **-.05%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.databaselookup.DatabaseLookup
 * Branch Change:  + 1.24%
 * Complexity Change:  + .77%
 * Instruction Change:  + .20%
 * Line Change:  + .67%

###### org.pentaho.di.trans.steps.dbproc.DBProcMeta
 * Instruction Change: **-.11%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.excelwriter.ExcelWriterStepMeta
 * Branch Change:  + 1.28%
 * Complexity Change:  + .78%
 * Instruction Change:  + .04%

###### org.pentaho.di.trans.steps.ifnull.IfNullMeta
 * Branch Change: **-2.00%**:small_red_triangle_down:
 * Complexity Change: **-1.89%**:small_red_triangle_down:
 * Instruction Change: **-.17%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.ldapoutput.LDAPOutputMeta
 * Instruction Change:  + .06%

###### org.pentaho.di.trans.steps.memgroupby.MemoryGroupByMeta
 * Branch Change:  + 2.35%
 * Complexity Change:  + 2.70%

###### org.pentaho.di.trans.steps.synchronizeaftermerge.SynchronizeAfterMergeMeta
 * Instruction Change:  + .04%

###### org.pentaho.di.trans.steps.tableoutput.TableOutputMeta
 * Branch Change:  + .91%
 * Complexity Change:  + .82%

###### org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta
 * Instruction Change:  + .18%

###### org.pentaho.di.trans.steps.gpload.GPLoadMeta
 * Instruction Change:  + .13%

###### org.pentaho.di.ui.repository.pur.repositoryexplorer.model.UISecurityRole
 * Branch Change: **-1.79%**:small_red_triangle_down:
 * Complexity Change: **-1.54%**:small_red_triangle_down:
 * Instruction Change: **-.16%**:small_red_triangle_down:
 * Line Change: **-.58%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.salesforceupdate.SalesforceUpdateMeta
 * Instruction Change: **-.17%**:small_red_triangle_down:

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'engine,core' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java
  core/src/test/java/org/pentaho/di/core/database/DatabaseUnitTest.java
  core/src/test/java/org/pentaho/di/core/row/value/ValueMetaBaseTest.java
  engine/src/main/java/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java
  engine/src/test/java/org/pentaho/di/trans/steps/databaselookup/DatabaseLookupUTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.di.core.row.value.ValueMetaBase
 * Instruction Change: **-.16%**:small_red_triangle_down:
 * Line Change: **-.29%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.evalfilesmetrics.JobEntryEvalFilesMetrics
 * Branch Change: **-.39%**:small_red_triangle_down:
 * Complexity Change: **-.51%**:small_red_triangle_down:
 * Instruction Change: **-.08%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.evaluatetablecontent.JobEntryEvalTableContent
 * Branch Change:  + 1.01%
 * Complexity Change:  + 1.06%
 * Instruction Change:  + .17%

###### org.pentaho.di.job.entries.waitforsql.JobEntryWaitForSQL
 * Branch Change: **-.66%**:small_red_triangle_down:
 * Complexity Change: **-.93%**:small_red_triangle_down:
 * Instruction Change: **-.13%**:small_red_triangle_down:

###### org.pentaho.di.trans.step.StepMeta
 * Branch Change: **-.68%**:small_red_triangle_down:
 * Complexity Change: **-.58%**:small_red_triangle_down:
 * Instruction Change: **-.19%**:small_red_triangle_down:
 * Line Change: **-.26%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.calculator.CalculatorMetaFunction
 * Branch Change:  + 8.70%
 * Complexity Change:  + 3.03%
 * Instruction Change:  + .75%
 * Line Change:  + .89%

###### org.pentaho.di.trans.steps.combinationlookup.CombinationLookupMeta
 * Instruction Change:  + .04%

###### org.pentaho.di.trans.steps.constant.ConstantMeta
 * Instruction Change:  + .10%

###### org.pentaho.di.trans.steps.databaselookup.DatabaseLookup
 * Branch Change:  + 1.24%
 * Complexity Change:  + .77%
 * Instruction Change:  + .20%
 * Line Change:  + .67%

###### org.pentaho.di.trans.steps.datagrid.DataGridMeta
 * Branch Change:  + 1.92%
 * Complexity Change:  + 1.69%
 * Instruction Change:  + .17%

###### org.pentaho.di.trans.steps.dimensionlookup.DimensionLookupMeta
 * Branch Change:  + .31%
 * Instruction Change:  + .06%
 * Line Change: **-.23%**:small_red_triangle_down:
 * Method Change: **-1.02%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.groupby.GroupByMeta
 * Branch Change: **-1.04%**:small_red_triangle_down:
 * Complexity Change: **-1.05%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.ifnull.IfNullMeta
 * Branch Change:  + 2.00%
 * Complexity Change:  + 1.89%
 * Instruction Change:  + .17%

###### org.pentaho.di.trans.steps.memgroupby.MemoryGroupByMeta
 * Branch Change:  + 2.35%
 * Complexity Change:  + 2.70%
 * Instruction Change:  + .18%
 * Line Change:  + .48%

###### org.pentaho.di.trans.steps.replacestring.ReplaceStringMeta
 * Instruction Change:  + .06%

###### org.pentaho.di.trans.steps.singlethreader.SingleThreaderMeta
 * Instruction Change:  + .10%

###### org.pentaho.di.trans.steps.synchronizeaftermerge.SynchronizeAfterMergeMeta
 * Instruction Change: **-.04%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.tableoutput.TableOutputMeta
 * Branch Change: **-.91%**:small_red_triangle_down:
 * Complexity Change: **-.82%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.webservices.WebServiceMeta
 * Branch Change: **-1.79%**:small_red_triangle_down:
 * Complexity Change: **-1.19%**:small_red_triangle_down:
 * Instruction Change: **-.07%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.googleanalytics.GaInputStepMeta
 * Branch Change: **-2.00%**:small_red_triangle_down:
 * Complexity Change: **-1.10%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.salesforceinsert.SalesforceInsertMeta
 * Instruction Change: **-.15%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.salesforceupsert.SalesforceUpsertMeta
 * Instruction Change:  + .14%

AliaksandrShuhayeu commented:
@mbatchelor Could you please review it?
All these strings were added in scope of PDI-8373. All these strings are a workaround in scope of PDI-8373.
Object[] ret = getRow( res, lazyConversion );
We could get second argument yes only in org.pentaho.di.trans.steps.databaselookup.DatabaseLookup L=141 . I removed it as not actual workaround.
I deleted my changes in Database.java. But these strings are not used in kettle more.
